src/gpuarray/util.h

```diff
@@ -98,6 +98,90 @@ GPUARRAY_PUBLIC void gpuarray_elemwise_collapse(unsigned int n,
                                                 unsigned int *nd,
                                                 size_t *dim, ssize_t **strs);
 
+
+typedef struct _ga_half_t { uint16_t h; } ga_half_t;
+
+/* code strongly inspired from
+   https://github.com/numpy/numpy/blob/master/numpy/core/src/npymath/halffloat.c#L246 */
+
+static inline ga_half_t ga_float2half(float f) {
```

Comment:
This seems to cause issues for Visual Studio. Actually only the most recent version of MSVC seems to be ok. Please see this [failed build]( https://ci.appveyor.com/project/conda-forge/libgpuarray-feedstock/build/1.0.9/job/fcetfa7a6ux8bfe3#L388 ) for details. Also this [SO question]( https://stackoverflow.com/q/32851292 ) and its [answer]( https://stackoverflow.com/a/32959521 ) seem pertinent.